src/routes/user/userComponents/userPlanData.svelte

```diff
@@ -1,14 +1,151 @@
 <script>
+    // Framework imports
+    import { getToastStore } from "@skeletonlabs/skeleton";
+    import { onMount } from "svelte";
+
+    // Import backend urls and fetch function
+    import { urls, tryFetchingRestricted } from "$lib/util.js";
+
     // Definitions
-    export let userID;
-    // ToDo logic for backend fetch
+    const toastStore = getToastStore();
+
+    // Export data from parent component
+    export let personalData;
+
+    let lastplanValue;
+    let formEditEnabled = false;
+    let planReference;
+    let planData = new Promise((resolve, reject) => {});
+
+    // Success toast
+    const successToast = {
+        message: "Deine Daten wurden erfolgreich aktualisiert",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-secondary",
+    };
+
+    // Warning toast
+    const warningToast = {
+        message: "Du hast keine zu aktualisierenden Daten eingegeben",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-warning",
+    };
+
+    // Error toast
+    const errorToast = {
+        message: "Deine Daten konnten nicht aktualisiert werden",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-error",
+    };
+
+    // Functions
+    function checkPlanNameDidntChange(planValue) {
+        return planValue == lastplanValue;
+    }
+
+    function resolvePlanNameToPlanId(planName) {
+        for (let index = 0; index < planData.length; index++) {
+            if (planData[index].name == planName) {
+                return planData[index].id;
+            }
+        }
+        throw new Error("Planname is not in backend plan list");
```

Comment:
```suggestion
        let plan = planData.find(plan => plan.name == planName);
        if (plan === undefined) {
            throw new Error("Planname is not in backend plan list");
        }
        return plan.id; 
```
Das sollte so auch etwas schöner gehen, bitte ausprobieren. Eventuell auch nochmal `npm run format` machen.